src/zcash/address/zip32.cpp

```diff
@@ -153,13 +297,73 @@ SaplingExtendedFullViewingKey SaplingExtendedSpendingKey::ToXFVK() const
     return ret;
 }
 
-libzcash::SaplingPaymentAddress SaplingExtendedSpendingKey::DefaultAddress() const
-{
-    return ToXFVK().DefaultAddress();
+//
+// Unified
+//
+
+std::optional<std::pair<UnifiedSpendingKey, HDKeyPath>> UnifiedSpendingKey::ForAccount(const HDSeed& seed, uint32_t bip44CoinType, uint32_t accountId) {
+    UnifiedSpendingKey usk;
+    usk.accountId = accountId;
+
+    auto transparentKey = DeriveZip32TransparentAccountKey(seed, bip44CoinType, accountId);
+    if (!transparentKey.has_value()) return std::nullopt;
+    usk.transparentKey = transparentKey.value().first;
+
+    auto saplingKey = SaplingExtendedSpendingKey::ForAccount(seed, bip44CoinType, accountId);
+    usk.saplingKey = saplingKey.first;
+
+    return std::make_pair(usk, saplingKey.second);
```

Comment:
How is the `HDKeyPath` returned here used? Should I expect to see an equivalent path returned for Orchard (and if so, what should be returned here for transparent)?

Reply:
The returned HDKeyPath is included in the CKeyMetadata. So yes, it should also be returned for Orchard and transparent keys.

Reply:
There's actually a wrinkle here with using the BIP 44 path - I had hoped that for unified keys, that we'd be able to have a single CKeyMetadata value with a consistent keypath between all of the components of the key, but since we're using `44'` as the `purpose` field for transparent keys and `32'` as the `purpose` field for Sapling/Orchard keys, we can't do that.

Reply:
The thing that is consistent between keys in our ZIP 316 HD scheme is the account number. The rest of the keypath is intentionally consistent with the existing per-protocol paths for compatibility. If we had wanted the full keypath to be identical, we'd have needed to discuss this much earlier; in particular, before auto-shielding was deployed to mobile wallets, which uses the BIP 44 path.

The way I would handle this in the `CKeyMetadata` is that each key within a USK has its own path as now, and in this function we either return the full paths, or (more likely) return a structure that contains the account number and can regenerate the corresponding path for each protocol type.

Reply:
Having a separate key metadata type for USKs makes sense, I'll take that route.